src/chunk.ts

```diff
@@ -46,11 +46,10 @@ export function chunk(): unknown {
 }
 
 function _chunk<T>(array: ReadonlyArray<T>, size: number): Array<Array<T>> {
-  const ret: Array<Array<T>> = Array.from({
-    length: Math.ceil(array.length / size),
-  });
-  for (let index = 0; index < ret.length; index++) {
-    ret[index] = array.slice(index * size, (index + 1) * size);
+  const length = Math.ceil(array.length / size);
+  const ret: Array<Array<T>> = Array.from({ length });
+  for (let index = 0, offset = 0; index < length; index++) {
+    ret[index] = array.slice(offset, (offset += size));
```

Comment:
It looks like this is the optimal solution. It's probably better because the array's initialization is expensive, as v8 probably needs to reset the user-space memory, which it might not do when simply pushing into the array and enlarging it only internally.

All the other optimizations yield marginal (1-3% improvements).

```suggestion
  const ret: Array<Array<T>> = [];
  for (let offset = 0; offset < array.length; offset += size) {
    ret.push(array.slice(offset, offset + size));
```